File: tests/generator/resources/algebraic_eqn_state_var_on_rhs_one_component/code.c
#include <math.h>

void initializeConstants(double *states, double *variables)
{
    states[0] = 1.0;
    variables[0] = 1.0;
}

void computeComputedConstants(double *variables)
{
}

void computeRates(double voi, double *states, double *rates, double *variables)
{
    rates[0] = variables[0];
}

void computeVariables(double voi, double *states, double *rates, double *variables)
{
    variables[1] = 2.0*states[0];
}

void computeStateRateBasedVariables(double voi, double *states, double *rates, double *variables)
{
    variables[1] = 2.0*states[0];
}

File: tests/resources/generator/ode_multiple_odes_with_same_name/code.py
from math import *


def initialize_constants(states, variables):
    states[0] = 1.0
    states[1] = 1.0
    variables[0] = 1.0


def compute_computed_constants(variables):
    pass


def compute_rates(voi, states, rates, variables):
    rates[0] = 1.0
    rates[1] = variables[0]


def compute_variables(voi, states, rates, variables):
    pass
